dso-l2/src/main/java/com/tc/objectserver/handler/ReplicatedTransactionHandler.java

```diff
@@ -282,17 +285,21 @@ private void replicatedActivityReceived(ServerID activeSender, SyncReplicationAc
 
     // Send the RECEIVED ack before we run this.
     ackReceived(activeSender, activity);
```

Comment:
I think that the order is wrong here:  We need to wait on the future before sending the ackReceived, not the ack for complete.